packages/react-core/src/components/Banner/Banner.tsx

```diff
@@ -13,29 +17,48 @@ export interface BannerProps extends React.HTMLProps<HTMLDivElement> {
    * be passed in when the banner conveys status/severity.
    */
   screenReaderText?: string;
-  /** Variant styles for the banner. */
-  variant?: 'default' | 'blue' | 'red' | 'green' | 'gold';
+  /** Color options for the banner, will be overwritten by any applied using the status prop. */
+  color?: BannerColor;
+  /** Status style options for the banner, will overwrite any color applied using the color prop. */
+  status?: BannerStatus;
+}
+interface StatusBanner extends BannerProps {
+  color?: never;
+  status?: BannerStatus;
 }
 
-export const Banner: React.FunctionComponent<BannerProps> = ({
+interface NonStatusBanner extends BannerProps {
+  color?: BannerColor;
+  status?: never;
+}
+
+export const Banner: React.FunctionComponent<StatusBanner | NonStatusBanner> = ({
```

Comment:
Tried this out as a way of preventing both `color` and `status` props at the type level, I'm not committed to it if we don't like it though.

Reply:
Not entirely sure how I feel about this. It's definitely an improvement from what was attempted in https://github.com/patternfly/patternfly-react/pull/8279/commits/91d79076009ffc6c0dfa22d69f272655b8e246a3 (with the main issue being it didn't render props in the prop table correctly, whereas this PR does). 

At the same time I'm almost wondering if it'd be better to just keep the `variant` prop, and makes its type `BannerStatus | BannerColor`?

Otherwise this looks good. If anything we could have a followup discussion about it since it's only going into the v6 branch so I wouldn't block over this going in now.